app/views/dashboard/index.html.erb

```diff
@@ -17,7 +17,9 @@
               <a class='minimize'>&minus;</a>
             </div>
             <div class="widget <%= widget.name.parameterize %>">
-              <%= render_widget(widget) %>
+              <div data-ajax-url="<%= widget_path(widget) %>" data-on-complete="refreshCharts">
+                <%= spinner _("Loading widget...") %>
```

Comment:
How about displaying the name of what's being loaded, next to a spinner? e.g: `spinner _(widget.name)`

I looked up the names of all widgets and they seem to make sense, so it'd at least give some info about what's loading in case it takes more time than usual or whatever:

```
["Status table", "Status chart", "Report summary", "Distribution chart", "Discovery widget", "Status table", "Status chart", "Report summary", "Distribution chart", "Discovery widget", "Status table", "Status chart", "Report summary", "Distribution chart", "Discovery widget", "Status table", "Status chart", "Report summary", "Distribution chart", "Discovery widget", "Status table", "Status chart", "Report summary", "Distribution chart", "Tasks Status table", "Tasks in Error/Warning", "Discovery widget", "Status table", "Status chart", "Report summary", "Distribution chart", "Tasks Status table", "Tasks in Error/Warning", "Discovery widget", "Tasks Status table", "Tasks in Error/Warning"]
```